lib/zip/entry.rb

```diff
@@ -678,6 +698,24 @@ def data_descriptor_size
       (@gp_flags & 0x0008) > 0 ? 16 : 0
     end
 
+    # For DEFLATED compression *only*: set the general purpose flags 1 and 2 to
+    # indicate compression level. This seems to be mainly cosmetic but they are
+    # generally set by other tools - including in docx files.
```

Comment:
The comments and link to APPNOTE in the PR description were helpful --- maybe worth adding them here?

Reply:
Done.